src/uart_comm_unix.c

```diff
@@ -360,6 +360,92 @@ static int uart_config_flowcontrol(int fd, const struct uart_config *config)
     return tcsetattr(fd, TCSANOW, &options);
 }
 
+int uart_get_rs485_config(struct uart *port, struct uart_config *config)
+{
+#if defined(__linux__)
+    struct serial_rs485 rs485;
+
+    if (ioctl(port->fd, TIOCGRS485, &rs485) < 0) {
+        // This may fail with ENOTTY if the port doesn't support RS485. If thats
+        // the case, then we only care about a failure when a user is trying
+        // to set things. Otherwise, we assume RS485 may not even be expected and
+        // can safely be ignored
+        if (!config->rs485_user_configured) return 0;
+
+        record_errno();
+        return -1;
+    }
+
+    config->rs485_enabled = (rs485.flags & SER_RS485_ENABLED);
```

Comment:
```suggestion
    config->rs485_enabled = (rs485.flags & SER_RS485_ENABLED) != 0;
```
I think that this needs to be `0` or `1` based on some other code.